### ISSUE TITLE
Setting a natGatewayId on a subnet causes segfaults

### ISSUE THREAD

**Issue by liztio** [kubernetes-sigs/cluster-api-provider-aws#1175]
/kind bug

**What steps did you take and what happened:**
I was trying to work arround #1173, so I set one subnet's natGatewayId to another's existing natGateway. Then I got a bunch of segfaults:

https://gist.github.com/liztio/6a336cbaec7e85d76e5b54f019a27c04

**What did you expect to happen:**
Some error messages at worst, an actual successful gateway provision at best

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: master
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by vincepri** [comment]
IIRC this is intentional and should be a read-only field, maybe we should make it clear for users

**Comment by liztio** [comment]
/assign
/lifecycle active

gonna fix this while I remember the context

**Comment by liztio** [comment]
/remove-lifecycle active

The easiest solution from my perspective is a check to make sure `natGatewayId` isn't set unless id is, and it makes sense to wait for #1116.

**Comment by vincepri** [comment]
Actually before I do this, are documentation on the fields and create webhook errors sufficient to solve this?

**Comment by aaroniscode** [comment]
Did some digging in the code and some testing here and I'm not sure this is an issue any more:

The original issue was opened when trying to work around the issue that only 1 NAT gateway was created. Now, anytime a public subnet is created, a NAT gateway is created.

I tried:
* Creating a new public subnet with a `natGatewayId` of another subnet, but a new NAT gateway was created and the ID of that NAT gateway was updated on the Subnet object.

* Edited a public subnet and changed the `natGatewayId` to another NatGatewayID, but during reconciliation the Subnet was updated with the correct `natGatewayId` 

* Setting a `natGatewayId` on a private subnet, but it is removed when reconciled.

So we might be OK to close this as no longer an issue. Let me know if I missed an error case.
